app/assets/javascripts/discourse/app/lib/flag-targets/flag.js

```diff
@@ -0,0 +1,57 @@
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class Flag {
+  flaggingTopic() {
+    return false;
+  }
+
+  isEditable() {
+    return true;
+  }
+
+  createFlag(controller, opts) {
+    // an instance of ActionSummary
+    let postAction = this.postActionFor(controller);
+
+    let params = controller.get("selected.is_custom_flag")
+      ? { message: this.message }
+      : {};
+
+    if (opts) {
+      params = Object.assign(params, opts);
+    }
```

Comment:
A little bit afraid of the possible side effects of this code. It's possibly mutating stuff without being super explicit about it (this.message for example).